controller_apache2.yml

```diff
@@ -0,0 +1,49 @@
+- name: Copy config file
+  template: src=files/apache2_ports.conf dest=/etc/apache2/ports.conf
+
+- name: Enable apache modules
+  apache2_module: state=present name="{{ item }}"
+  with_items:
+    - proxy
+    - rewrite
+    - proxy_http
+    - headers
+    - ssl
+
+- name: Install jiocloud ssl certs
+  apt: pkg="{{ item }}" state=latest force=yes
+  with_items:
+    - ca-certificates
+    - jiocloud-ssl-certificate
+
+- name: Copy ssl certificate to etc
+  command: cp /etc/apache2/certs/jiocloud.com.crt /etc/ssl/certs/
+
+- name: Ensure ssl/keys folder
+  file: dest=/etc/ssl/keys state=directory
+
+- name: Copy ssl key to etc
+  command: cp /etc/apache2/keys/jiocloud.com.key /etc/ssl/keys/
+
+- name: Add crt to usr
+  command: cp /etc/ssl/certs/jiocloud.com.crt /usr/local/share/ca-certificates/selfsigned.crt
+
+- name: Refresh CA list
+  command: update-ca-certificates --fresh
+
+- name: Remove sites-available
```

Comment:
No need to remove complete folder. This will lead to deletion of files every time  playbooks run. Further,   ansible can just apply diff changes in vhost file. We can delete the default file if required.